src/types/types.js

```diff
@@ -1,3 +1,4 @@
+// Layer transformation
```

Comment:
What's this about?

Reply:
I added that to describe the types (I thought it might be useful when there are lots of types).

I'll just remove it for now and see what you say in response to this comment.

Reply:
Oh right. Yeah, for sure it can be useful to describe the types when there's a lot going on!

I think this particular case that comment could be misleading because the `Layer` type is (conceptually) used for more than just the transformations in `layers.js`. They appear in the document and get rendered by the canvas etc. You could actually argue that all the transformations in `layers.js` don't actually use the `Layer` type at all, but instead an inexact object with `x1`, `y1`, `x2` and `y2` properties:

```js
type TransformableObject = {
  x1: number,
  y1: number,
  x2: number,
  y2: number,
  ...
};
```

A layer just happens to conform to that type, but let's just leave it as is for now. 🙂

Reply:
Now that makes a lot of sense. I was unsure how to structure the type so that it must have certain fields but didn't have to have other fields. It was erroring because layer was being passed in but my type (originally) was just 
```
{
  x1: number,
  y1: number,
  x2: number,
  y2: number,
}
```